datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadbalancingStrategy.java

```diff
@@ -101,14 +101,19 @@ private List<DatastreamTask> getDatastreamTasks(List<DatastreamGroup> datastream
     List<DatastreamTask> currentlyAssignedDatastreamTasks = new ArrayList<>();
     currentAssignment.values().forEach(currentlyAssignedDatastreamTasks::addAll);
 
+    Set<String> uniqueDatastreamTaskNames = new HashSet<>();
```

Comment:
making this variable local to datastream group will eliminate the need of added code on line 113.

Also, can you rename with set

Reply:
Is there any scenario where we could have a duplicate DatastreamGroup? It looks to be a list and not a set